model/page.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+require_relative "../model"
+
+require "pagerduty"
+
+class Page < Sequel::Model
+  dataset_module do
+    def active
+      where(resolved_at: nil)
+    end
+  end
+
+  def after_create
+    return unless Config.pagerduty_key
+
+    @@pagerduty ||= Pagerduty.build(integration_key: Config.pagerduty_key, api_version: 2)
```

Comment:
I think we need move it outer place that `resolve` doesn't depend `after_create`. `after_create` and `resolve` might be called in different clover processes, for example respirate created page and web console resolve it or respirate restarted and @@pagerduty gone

Reply:
Page almost certainly needs a strand. Resolution would be via semaphore.

Reply:
I created a method called pagerduty_client to cache the client instead of moving it to strand. I didn't want to couple pages and strands yet. Coupling deadlines and strands is fine (which we did in the PR), but I suspect we might want to create pages in other contexts (a cronjob maybe?, not sure just speculating).  Also keeping the paging related logic in one file felt better.

If you still prefer moving client to strands, that can be done as well.

Reply:
where the client gets instantiated is not such a big deal, the question is how do we move through the lifecycle for a page, which is in every respect much like moving through the lifecycle of a VM, except simpler.
 
For example, the POST may succeed, but you get an IOError because of an unlucky connection drop. Transaction aborts, but a page that doesn't get auto-resolved (because it has no record) is in fact in PagerDuty.

If you make a Page.assemble that includes making a strand, the `start` label can try+retry, and because all you do in the synchronous path for starting the deadline is create two records, with no other side effects.